sktime/annotation/e_agglo.py

```diff
@@ -0,0 +1,435 @@
+# -*- coding: utf-8 -*-
+"""E-Agglo: agglomerative clustering algorithm that preserves observation order."""
+
+import warnings
+
+import numpy as np
+from scipy.spatial.distance import cdist
+
+from sktime.transformations.base import BaseTransformer
+
+__author__ = ["KatieBuc"]
+__all__ = ["EAGGLO"]
+
+
+class EAGGLO(BaseTransformer):
+    """
+    Hierarchical agglomerative estimation of multiple change points, outlined in [1]_.
+
+    E-Agglo is a non-parametric clustering approach for multivariate timeseries, where
+    neighboring segments are sequentially merged to maximize a goodness-of-fit
+    statistic. Unlike most general purpose agglomerative clustering algorithms, this
+    procedure preserves the time ordering of the observations.
+
+    This method can detect any distributional change within an independent sequence,
+    and does not make any distributional assumptions (beyond the existence of an
+    alpha-th moment). Estimation is performed in a manner that simultaneously
+    identifies both the number and locations of change points.
+
+    Parameters
+    ----------
+    member : array_like (default=None)
+        1D `array_like` representing the initial cluster membership for input
+        data, X.
+    alpha : float (default=1.0)
+        fixed constant alpha in (0, 2) used in the divergence measure, as the
```

Comment:
based on `assert self.alpha > 0 and self.alpha <= 2` from `_fit` this should be
```suggestion
        fixed constant alpha in (0, 2] used in the divergence measure, as the
```